### PR TITLE
Split aya-persist into aya-write-yasnippet and aya-build-yasnippet

### PR DESCRIPTION
**Pull request by jasonm23**
Note this is incomplete, README will need to be updated when the features are ratified.

I expect both have room to change and are separate commands in their own right.

### CODE REVIEW

File: auto-yasnippet.el [hunk 1]
```diff
@@ -108,7 +108,8 @@
 (defcustom aya-persist-snippets-dir
   "~/.emacs.d/snippets"
   "Directory to save auto yasnippets."
-  :type 'directory)
+  :type 'directory
```
Comment (abo-abo) [inline auto-yasnippet.el]: There's no need to add 'group, it's automatic.

**Comment by abo-abo** [comment]
There's too much copy pasting, I don't like it.

Here's how I expect you to write it:

```cl
(defun aya-write-yasnippet ()
  (interactive)
  (let ((aya-insert-snippet-function
         'aya-insert-snippet-function-extra))
    (call-interactively 'aya-persist-snippet)))
```

You can add as many parameters to `aya-insert-snippet-function-extra`
as you want.

**Comment by jasonm23** [comment]
Back from vacation, I'll get this organised soon.

**Comment by jasonm23** [comment]
Let's kill this off, I think it's better as a separate package which depends on auto-snippet.

That way the intent of auto-yasnippet as a way to create **temporary** yasnippets isn't diluted.

I can also then build the feature however I feel like, since I think I'm the guy who actually wants it.

**Comment by abo-abo** [comment]
Good idea.

**Pull request closed without merging**